File: Ninja Training/Recursion.java

class Recursion {

    public int NinjaJump(int day, int last, int[][] arr) {
        if (day == 0) {
            int maximum = 0;
            for (int i = 0; i < 3; i++) {
                if (i != last) {
                    int temp = arr[day][i];
                    maximum = Math.max(maximum, temp);
                }
            }

            return maximum;
        }

        int maximum = 0;
        for (int i = 0; i < 3; i++) {
            if (i != last) {
                int temp = arr[day][i] + NinjaJump(day - 1, i, arr);
                maximum = Math.max(maximum, temp);
            }
        }

        return maximum;
    }

    public static void main(String[] args) {
        int[][] arr = {
            {18, 11, 19},
            {4, 13, 7},
            {1, 8, 13}
        };

        int size = arr.length - 1;
        int maximum = 0;

        Recursion obj = new Recursion();

        for (int i = 0; i < 3; i++) {
            int temp = arr[size][i] + obj.NinjaJump(size - 1, i, arr);
            maximum = Math.max(maximum, temp);
        }

        System.out.println("Maximum ninja jump is : " + maximum);
    }
}

File: Ninja Training/Recursion.py
def NinjaTrain(i,j,arr):
    if i == 0:
        maximum = 0
        for k in range(3):
            if j != k:
                maximum = max(maximum,arr[0][k])
        return maximum

    maximum = 0
    for k in range(3):
        if j != k:
            maximum = max(maximum,arr[i][k] + NinjaTrain(i-1,k,arr) )
    
    return maximum
  
def ninjaTraining(n, arr):

    maximum = 0
    for j in range(3):
            dummy_max = arr[n-1][j] + NinjaTrain(n-2,j,arr)
            maximum = max(maximum, dummy_max )

    return maximum

if __name__ == "__main__":
    print(ninjaTraining(3,[ [18 ,11 ,19],
                            [4, 13 ,7],
                            [1, 8 ,13] ] ))